router/core/flushwriter.go

```diff
@@ -21,8 +31,11 @@ type HttpFlushWriter struct {
 	flusher       http.Flusher
 	subscribeOnce bool
 	sse           bool
+	multipart     bool
```

Comment:
We should start to encapsulate SSE and multipart in different writer implementations.